authors/apps/authentication/tests/test_jwt.py

```diff
@@ -0,0 +1,176 @@
+    def token(self):
+        return self.login_a_user()
+
+    def test_existance_of_jwt(self):
+        """
+        Test to check if a token is actually generated on successful user registration
+        """
+
+        response = self.test_client.post(
+            "/api/users/login/", data=json.dumps(self.data), content_type='application/json')
+
+        decoded_token = jwt.decode(
+            response.json()['user']['token'], authors.settings.base.SECRET_KEY, 'HS256')
+
+        self.assertTrue("token" in response.json()['user'])
+        self.assertEqual(decoded_token['username'], self.username)
+
+    def test_to_check_for_valid_token(self):
+        """
+        Test to check if a token is passed is valid the endpoint authorises the user
+        """
+        headers = {'HTTP_AUTHORIZATION': 'Token ' + self.token}
+
+        response = self.test_client.get(
+            "/api/user/", **headers, content_type='application/json')
+
+        self.assertEqual(response.status_code, 200)
+
+    def test_to_check_for_in_valid_token(self):
+        """
+        Test to check if a token passed is invalid
+        """
+
+        headers = {
+            'HTTP_AUTHORIZATION': 'Token ' + self.token + 'werfasgs'
+        }
+
+        response = self.test_client.get(
+            "/api/user/", **headers, content_type='application/json')
+        print(response.json())
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response.json()[
+                         'user']['detail'], "Invalid authentication. Could not decode token.")
+
+    def test_no_actual_token_is_passed_in_header(self):
+        """
+        Test to check if no actual token is passed is valid the endpoint authorises the user
+        """
+
+        headers = {
+            'HTTP_AUTHORIZATION': 'Token '
+        }
+
+        response = self.test_client.get(
+            "/api/user/", **headers, content_type='application/json')
+        print(response)
+        self.assertEqual(response.status_code, 401)
+
+    def test_more_than_token_passed_in_header(self):
+        """
+        Test to check if a token is passed is valid the endpoint authorises the user
+        """
+
+        headers = {
+            'HTTP_AUTHORIZATION': 'Token ' + self.token + ' somethingelse'
+        }
+
+        response = self.test_client.get(
+            "/api/user/", **headers, content_type='application/json')
+        print(response)
+        self.assertEqual(response.status_code, 401)
+
+    def test_wrong_prefix_passed_in_header(self):
+        """
+        Test to check when a wrong prefix is passed in the header
+        """
+
+        headers = {
+            'HTTP_AUTHORIZATION': 'Bear ' + self.token
+        }
+
+        response = self.test_client.get(
+            "/api/user/", **headers, content_type='application/json')
+        print(response)
+        self.assertEqual(response.status_code, 401)
+
+    def test_token_passed_but_user_doesnt_exist(self):
```

Comment:
@lytes20 
- This test actually seems to do what the one on line 69 is doing
- Testing for a valid token